lib/appmap/rails/sql_handler.rb

```diff
@@ -71,21 +71,32 @@ def execute_query(sql)
           end
         end
 
-        class ActiveRecordExaminer
-          def server_version
-            ActiveRecord::Base.connection.try(:database_version) ||\
-              warn("Unable to determine database version for #{database_type.inspect}")
-          end
-
-          def database_type
-            type = ActiveRecord::Base.connection.adapter_name.downcase.to_sym
-            type = :postgres if type == :postgresql
+        if defined?(ActiveRecord)
+          class ActiveRecordExaminer
+            class << self
+              attr_reader :server_version, :database_type
+              
+              @database_type = begin
+                type = ActiveRecord::Base.connection.adapter_name.downcase.to_sym
+                type = :postgres if type == :postgresql
+                
+                type
+              end
+              
+              @server_version = ActiveRecord::Base.connection.try(:database_version) or
```

Comment:
Will this work if the app is using multiple different connections?

Reply:
Nope. I'll figure out a different approach that handles that case, too.